rust/lancedb/src/table.rs

```diff
@@ -1870,12 +1919,22 @@ impl TableInternal for NativeTable {
                 }
             }
 
-            let query_vector = query_vector.as_primitive::<Float32Type>();
-            scanner.nearest(
-                &column,
-                query_vector,
-                query.base.limit.unwrap_or(DEFAULT_TOP_K),
-            )?;
+            if is_binary {
+                let query_vector = arrow::compute::cast(&query_vector, &DataType::UInt8)?;
```

Comment:
is query_vector here already forced to be a float32? I worry about the accuracy of it here...

Reply:
yeah it's float32 here.
the values must be integers in range 0~255 (even stored as float32s) so it won't lose any accuracy?
I just added a test to verify this